packagedb/management/commands/fix_purl_values.py

```diff
@@ -0,0 +1,226 @@
+    contents = response.json()
+    resp = contents.get('response', {})
+    matched_artifacts = []
+    if resp.get('numFound', 0) > 0:
+        for matched_artifact in resp.get('docs', []):
+            namespace = matched_artifact.get('g', '')
+            name = matched_artifact.get('a', '')
+            version = matched_artifact.get('v', '')
+            ec = matched_artifact.get('ec', [])
+            if not namespace and name and version:
+                continue
+            matched_artifacts.append(
+                MavenArtifact(
+                    namespace=namespace,
+                    name=name,
+                    version=version,
+                    ec=ec,
+                )
+            )
+    return matched_artifacts
+
+
+class Command(VerboseCommand):
+    help = 'Update maven Package download_url values'
+
+    def handle(self, *args, **options):
+        maven_packages = Package.objects.filter(type='maven', sha1__is_null=False)
+        maven_packages_count = maven_packages.count()
+        logger.info(f'Checking {maven_packages_count:,} Maven Package PackageURL values')
+        packages_to_delete = []
+        unsaved_packages = []
+
+        processed_packages_count = 0
+        for i, package in enumerate(MemorySavingQuerysetIterator(maven_packages)):
+            matched_artifacts = query_sha1_on_maven(package.sha1)
+            if not matched_artifacts:
+                # Remove this package from the database because it's not on maven
+                packages_to_delete.append(package)
+            for artifact in matched_artifacts:
+                artifact_namespace = artifact.namespace
+                artifact_name = artifact.name
+                artifact_version = artifact.version
+                artifact_qualifiers = artifact.qualifiers
+
+                package_different_case = Package.objects.get_or_none(
+                    namespace__iexact=artifact_namespace,
+                    name__iexact=artifact_name,
+                    version__iexact=artifact_version,
+                    artifact_qualifiers__iexact=artifact_qualifiers,
+                )
+
+                # check to see if the artifact matches the current package
+                if (
+                    package.namespace == artifact_namespace
+                    and package.name == artifact_name
+                    and package.version == artifact_version
+                    and package.qualifiers == artifact_qualifiers
+                ):
+                    # check download url
+                    if requests.head(package.download_url):
+                        # Continue if it resolves
+                        continue
+                    else:
+                        package.download_url = artifact.download_url
+                        package.release_date = artifact.release_date
+                        package.save()
+                elif package_different_case:
```

Comment:
We should also update the SHA1 field for `package_different_case`. There can be cases where the SHA1 is either missing or incorrect for `package_different_case`.